presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java

```diff
@@ -132,13 +133,68 @@ public int getSetBits(int batchSize, boolean[] vector)
             throws IOException
     {
         int count = 0;
-        for (int i = 0; i < batchSize; i++) {
+        int bitsInFirstByte = Math.min(batchSize, bitsInData);
+        for (int i = 0; i < bitsInFirstByte; i++) {
+            vector[i] = nextBit();
+            count += vector[i] ? 1 : 0;
+        }
+        verify(bitsInData == 0);
+        int wholeBytes = (batchSize - bitsInFirstByte) / 8;
+        int numFilled = bitsInFirstByte;
+        for (int i = 0; i < wholeBytes; i++) {
+            byte data = byteStream.next();
+            count += Integer.bitCount(data);
+            vector[numFilled] = (data & 0x80) != 0;
+            vector[numFilled + 1] = (data & 0x40) != 0;
```

Comment:
nit: this change doesn't really belong to this PR; would you submit it to master?